Source/EmbeddedBoundary/WarpXInitEMField.cpp

```diff
@@ -0,0 +1,76 @@
+
+#include "WarpX.H"
+#include <cmath>
+
+amrex::RealArray
+WarpX::AnalyticSolSphere (amrex::Real x, amrex::Real y, amrex::Real z, amrex::Real t, amrex::Real r_sphere) {
+    amrex::Real k=2.7437/r_sphere;
+      amrex::Real r = sqrt(pow(x, 2) + pow(y, 2) + pow(z, 2));
+      amrex::Real theta = atan2(sqrt(pow(x, 2) + pow(y, 2)), z);
+      amrex::Real phi = atan2(y, x);
+      amrex::Real H_r = 0;
+      amrex::Real H_theta = 0;
+      amrex::Real mu_r = PhysConst::mu0;
+    amrex::Real H_phi = k / (r_sphere * mu_r) * std::sph_bessel(1, k*r) * sin(theta) * cos(k * PhysConst::c * t);
```

Comment:
We decided to already require C++17 for EB features (instead of pulling in external dependencies).
memo to myself: reflect that in build systems and add checks at compile time asserting `__cplusplus` version.